### PR TITLE
Add auto remove flag

### PR DESCRIPTION
**Pull request by dvic**
This PR adds the auto_remove flag (enabled by default). This flag signals docker to remove the container when it's stopped. In this way, when you don't use the reaper you at least don't keep all the successfully stopped containers.

### CODE REVIEW

**Comment by dvic** [comment]
I bumped `erlef/setup-beam` to v1, I assume this will resolve the CI issue.

File: test/excontainers/auto_remove_test.exs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    {:ok, container_id} = Docker.Containers.run(@sample_container)
+
+    assert container_exists?(container_id)
+
+    assert :ok = Docker.Containers.stop(container_id)
+
+    refute container_exists?(container_id)
+  end
+
+  defp container_exists?(container_id) do
```
Comment (dallagi) [inline test/excontainers/auto_remove_test.exs]: I'd consider extracting this into `test/support/docker_test_utils.ex` given the same helper function is also defined in `test/excontainers/resources_reaper_test.exs`, WDYT?

**Review by dallagi — COMMENTED** [review]
Thanks @dvic for the contribution!
This looks good, I only added a minor comment :)

**Comment by dallagi** [comment]
BTW I fixed the CI on master, feel free to rebase from there.
Just be aware that for some reason some test seems to be a little flaky now - I haven't figured out why yet.

File: test/excontainers/auto_remove_test.exs (hunk 1 above)
Comment (dvic) [reply test/excontainers/auto_remove_test.exs]: Sounds good!

**Comment by dvic** [comment]
> BTW I fixed the CI on master, feel free to rebase from there.
> Just be aware that for some reason some test seems to be a little fleaky now - I haven't figured out why yet.

Ah ok, I merged master inside this PR, if you want a rebase and squash, let me know, and I'll rebase and squash the commits.

**Comment by dallagi** [comment]
> Ah ok, I merged master inside this PR, if you want a rebase and squash, let me know, and I'll rebase and squash the commits.

No worries, it's going to be squashed on merge anyway :)
One final thing, could you do a `mix format` run? Then I'll merge right away

**Comment by dvic** [comment]
> > Ah ok, I merged master inside this PR, if you want a rebase and squash, let me know, and I'll rebase and squash the commits.
> 
> No worries, it's going to be squashed on merge anyway :) One final thing, could you do a `mix format` run? Then I'll merge right away

done!

**Comment by dallagi** [comment]
Thank you!

**Pull request merged**